### PR TITLE
Add test for thread use case of fusion

### PR DESCRIPTION
**Pull request by kmaehashi**
Wrote test for #1581

### CODE REVIEW

**Comment by kmaehashi** [comment]
Jenkins, test this please.

**Comment by kmaehashi** [comment]
@niboshi Thanks, fixed.

**Comment by niboshi** [comment]
LGTM

**Comment by kmaehashi** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
LGTM.

**Comment by kmaehashi** [comment]
Jenkins, test this please.

**Comment by kmaehashi** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
LGTM.

File: tests/cupy_tests/core_tests/test_fusion.py [hunk 1]
```diff
@@ -1567,3 +1568,64 @@ def f(x):
...
+            out[0] = f(x, y)
+
+        t = threading.Thread(target=_target, args=(x, y))
+        t.daemon = True
+        t.start()
+        t.join()
+        assert (out[0] == f(x, y)).all()
+
+    @testing.numpy_cupy_array_equal()
+    def test_thread_multiple_dtypes(self, xp):
```
Comment (niboshi) [inline tests/cupy_tests/core_tests/test_fusion.py]: What's the intention of this test?

**Pull request merged**